include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -0,0 +1,1280 @@
+        }
+    }
+
+    /*!\brief Construct/assign with `count` times `value`.
+     * \tparam rng_type The type of range to be inserted; must be "like" `value_type`, i.e. satisfy
+     * seqan3::input_range_concept and have the same `value_type` as `value_type`.
+     * \param count Number of elements.
+     * \param value The initial value to be assigned.
+     *
+     * \par Complexity
+     *
+     * In \f$O(count)\f$.
+     *
+     * \par Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    template <typename rng_type>
+    concatenated_sequences(size_type const count, rng_type && value)
+        requires std::is_same_v<std::decay_t<rng_type>, std::decay_t<reference>> ||
+                 std::is_same_v<std::decay_t<rng_type>, std::decay_t<const_reference>> ||
+                 (input_range_concept<std::decay_t<rng_type>> &&
+                  std::is_same_v<std::decay_t<ranges::value_type_t<std::decay_t<rng_type>>>,
+                                 std::decay_t<ranges::value_type_t<std::decay_t<value_type>>>>)
+    {
+        // TODO SEQAN_UNLIKELY
+        if (count == 0)
+            return;
+
+        auto repeated = ranges::view::repeat_n(value, count)
+                      | ranges::view::join
+                      | ranges::view::bounded;
+        data_values.reserve(count * ranges::size(value));
+        data_values = repeated;
+
+        data_delimiters.resize(count + 1);
+        for (size_type i = 1; i <= count; ++i)
+            data_delimiters[i] = i * ranges::size(value);
+    }
+
+    /*!\brief Construct/assign from pair of iterators.
+     * \tparam begin_iterator_type Must satisfy seqan3::input_iterator_concept and have same the same value_type as
+     * `iterator`.
+     * \tparam end_iterator_type Must satisfy seqan3::sized_sentinel_concept.
+     * \param begin_it begin of range to construct/assign from.
+     * \param end_it end of range to construct/assign from.
+     *
+     * \par Complexity
+     *
+     * In \f$O(\f$end_it - begin_it\f$)\f$.
+     *
+     * \par Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    template <typename begin_iterator_type, typename end_iterator_type>
+    concatenated_sequences(begin_iterator_type begin_it, end_iterator_type end_it)
+        requires input_iterator_concept<begin_iterator_type> &&
+                 std::is_same_v<ranges::value_type_t<begin_iterator_type>, std::decay_t<value_type>>//&& sized_sentinel_concept<end_iterator_type, begin_iterator_type>
+    {
+        data_delimiters.reserve(end_it - begin_it);
+
+        while (begin_it != end_it)
+        {
+            push_back(*begin_it);
+            ++begin_it;
+        }
+    }
+
+    /*!\brief Construct/assign from `std::initializer_list`.
+     * \tparam rng_type The type of range to be inserted; must be "like `value_type`", i.e. satisfy
```

Comment:
Like I said, verbalising the requirements completely would make the description overly complex... but I will try to think of something.